pkg/api/types.go

```diff
@@ -2070,10 +2070,41 @@ const (
 	StrategicMergePatchType PatchType = "application/strategic-merge-patch+json"
 )
 
+// Type and constants for component type.
+type ComponentType string
+
+// These are the valid conditions for the component type.
+const (
+	ComponentAPIServer         ComponentType = "apiserver"
+	ComponentControllerManager ComponentType = "controller-manager"
+	ComponentScheduler         ComponentType = "scheduler" // TODO: move mesos-specific component type & validation to contrib
+)
+
+// Component describes metadata about a core component.
+type Component struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	// Type of the component
+	Type ComponentType `json:"type"`
+	// URL to reach the component (scheme, host, port, path)
+	URL string `json:"location"`
+
+	// The time at which the component was last updated.
+	LastTimestamp util.Time `json:"lastTimestamp,omitempty"`
```

Comment:
shouldn't this be status?  It's a weird object in that "spec" really has no meaning - does that mean we should buck the trend of spec/status?  I don't think so.